### PR TITLE
fix(ui): delete repository manager from UI

### LINKED ISSUE #4432: feat(ui): delete repository manager from UI
Today, it's possible to delete the repository manager if it's used by an application with cdsctl only.

This issue is to add a confirmation modal if the repository manager is used by an application on webui.

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

1. Description
1. Related issues
close #4432
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: ui/src/assets/i18n/en.json [hunk 1]
```diff
@@ -638,7 +638,10 @@
   "integration_storage": "storage",
   "queue_monitoring": "Queue Monitoring",
   "repo_name": "Repository",
-  "repoman_delete_msg_ok": "Repository mananger is detached from the project",
+  "repoman_delete_msg_ok": "Repository manager is detached from the project",
+  "repoman_delete_confirm_message": "Are you sure to delete this repository manager from your project ?",
```
Comment (yesnault) [inline ui/src/assets/i18n/en.json]:
```suggestion
  "repoman_delete_confirm_message": "Are you sure to delete this repository manager from your project?",
```

File: ui/src/assets/i18n/en.json [hunk 2]
```diff
@@ -638,7 +638,10 @@
...
+  "repoman_delete_dependencies_message": "Are you sure to delete this repository manager from your project ? It's linked to these applications: {{apps}}",
```
Comment (yesnault) [inline ui/src/assets/i18n/en.json]:
```suggestion
  "repoman_delete_dependencies_message": "Are you sure to delete this repository manager from your project? It's linked to these applications: {{apps}}",
```

File: ui/src/assets/i18n/fr.json [hunk 3]
```diff
@@ -639,6 +639,9 @@
   "queue_monitoring": "Queue Monitoring",
   "repo_name": "Nom du dépôt",
   "repoman_delete_msg_ok": "Le gestionnaire de dépôt n'est plus lié au projet",
+  "repoman_delete_confirm_message": "Êtes-vous sur de supprimer ce gestionnaire de dépot du projet ?",
```
Comment (yesnault) [inline ui/src/assets/i18n/fr.json]:
```suggestion
  "repoman_delete_confirm_message": "Êtes-vous sur de supprimer ce gestionnaire de dépôt du projet ?",
```

File: ui/src/assets/i18n/fr.json [hunk 4]
```diff
@@ -639,6 +639,9 @@
...
+  "repoman_delete_dependencies_message": "Êtes-vous sur de supprimer ce gestionnaire de dépot du projet ? Il est actuellement lié à ces applications: {{apps}}",
```
Comment (yesnault) [inline ui/src/assets/i18n/fr.json]:
```suggestion
  "repoman_delete_dependencies_message": "Êtes-vous sur de supprimer ce gestionnaire de dépôt du projet ? Il est actuellement lié à ces applications: {{apps}}",
```

File: engine/api/repositoriesmanager/dao.go [hunk 5]
```diff
@@ -167,3 +169,29 @@ func DeleteForApplication(db gorp.SqlExecutor, app *sdk.Application) error {
...
 	return nil
 }
+
+//LoadLinkedApplicationNames loads applications which are linked with this repository manager name
+func LoadLinkedApplicationNames(db gorp.SqlExecutor, projectKey, rmName string) (sdk.IDNames, error) {
+	query := `SELECT application.id, application.name, application.description
+	FROM application
+		JOIN project ON project.id = application.project_id
+	WHERE project.projectkey = $1 AND application.vcs_server = $2`
+	rows, err := db.Query(query, projectKey, rmName)
```
Comment (fsamin) [inline engine/api/repositoriesmanager/dao.go]: you should use gorp for scanning with the type `IDName`: https://github.com/ovh/cds/blob/5f39550667cd1151fdb00ab159b6652df45a82b7/sdk/common.go#L42

**Pull request merged**